src/main/java/org/damap/base/rest/dmp/mapper/MapperService.java

```diff
@@ -67,11 +68,14 @@ public InternalStorage getInternalStorageById(Long id) {
    * @return a {@link org.damap.base.rest.storage.InternalStorageDO} object
    */
   public InternalStorageDO getInternalStorageDOById(Long id, String languageCode) {
-    InternalStorageTranslation internalStorageTranslation =
-        internalStorageTranslationRepo.getInternalStorageById(id, languageCode);
-    if (internalStorageTranslation != null)
+
+    InternalStorage internalStorage = internalStorageRepo.findById(id);
```

Comment:
```suggestion
    InternalStorage internalStorage = internalStorageRepo.findById(id);
    if (internalStorage == null) {
      return null;
    }
```

Having the null check and return first allows to reduce logic and unnecessary calls later.

Reply:
Done